src/averageAges.js

```diff
@@ -20,6 +20,17 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const man = people.filter(person => person.sex === 'm');
+
+  const manAvgAge = century
+    ? man
+      .filter(person => Math.ceil(person.died / 100) === century)
+      .reduce((memo, person) => memo + (person.died - person.born), 0)
+      / man.filter(person => Math.ceil(person.died / 100) === century).length
+    : man.reduce((memo, person) => memo + (person.died - person.born), 0)
+      / man.length;
```

Comment:
Let's simplify this logic

In case if century is passed in this function - you'll iterate array 4 times (`filter`, another `filter`, `reduce`, and another `filter`) This amount can be less 🙂  Just let's filter array only once (and use ternary operator inside to get needed filter condition if century is passed or not)

And then calculate sum of ages with `reduce`

Reply:
Also you've used average value calculating in next function, so better move this logic to helper function and reuse it